src/Payments/VoucherDisplay.res

```diff
@@ -0,0 +1,110 @@
+open Utils
+@send external click: Dom.element => unit = "click"
+@react.component
+let make = () => {
+  let (openModal, setOpenModal) = React.useState(_ => false)
+  let (returnUrl, setReturnUrl) = React.useState(_ => "")
+  let (downloadUrl, setDownloadUrl) = React.useState(_ => "")
+  let (reference, setReference) = React.useState(_ => "")
+  let logger = Recoil.useRecoilValueFromAtom(RecoilAtoms.loggerAtom)
+  let (downloadCounter, setDownloadCounter) = React.useState(_ => 0)
+  let (paymentMethod, setPaymentMethod) = React.useState(_ => "")
+  let (paymentIntent, setPaymentIntent) = React.useState(_ => Js.Json.null)
+  let (loader, setLoader) = React.useState(_ => true)
+  let linkRef = React.useRef(Js.Nullable.null)
+
+  React.useEffect1(() => {
+    switch linkRef.current->Js.Nullable.toOption {
+    | Some(link) => link->click
+    | None => ()
+    }
+    None
+  }, [loader])
+
+  React.useEffect0(() => {
+    handlePostMessage([("iframeMountedCallback", true->Js.Json.boolean)])
+    let handle = (ev: Window.event) => {
+      let json = ev.data->Js.Json.parseExn
+      let dict = json->Utils.getDictFromJson
+      if dict->Js.Dict.get("fullScreenIframeMounted")->Belt.Option.isSome {
+        let metadata = dict->getJsonObjectFromDict("metadata")
+        let metaDataDict =
+          metadata->Js.Json.decodeObject->Belt.Option.getWithDefault(Js.Dict.empty())
+        setReturnUrl(_ => metaDataDict->getString("returnUrl", ""))
+        setDownloadUrl(_ => metaDataDict->getString("voucherUrl", ""))
+        setReference(_ => metaDataDict->getString("reference", ""))
+        setPaymentMethod(_ => metaDataDict->getString("paymentMethod", ""))
+        setPaymentIntent(_ => metaDataDict->getJsonObjectFromDict("payment_intent_data"))
+        setLoader(_ => false)
+      }
+    }
+    Window.addEventListener("message", handle)
+    Some(() => {Window.removeEventListener("message", handle)})
+  })
+
+  let closeModal = () => {
+    postSubmitResponse(~jsonData=paymentIntent, ~url=returnUrl)
+    Modal.close(setOpenModal)
+  }
+
+  <Modal showClose=false openModal setOpenModal loader>
+    <div className="flex flex-col h-full justify-between items-center">
+      <div className=" flex flex-col max-w-md justify-between items-center">
+        <div className="flex flex-row w-full">
+          <p className="Disclaimer font-medium text-sm text-[#151A1F] opacity-70">
```

Comment:
Is this className `Disclaimer` being used?

Reply:
This class was present in the qr modal as well, just reused it